src/libsync/syncengine.cpp

```diff
@@ -1179,6 +1180,34 @@ void SyncEngine::checkForPermission(SyncFileItemVector &syncItems)
             if ((*it)->_isDirectory) {
                 for (SyncFileItemVector::iterator it_next = it + 1; it_next != syncItems.end() && (*it_next)->_file.startsWith(path); ++it_next) {
                     it = it_next;
+                    if ((*it)->_direction != SyncFileItem::Up && (*it)->_instruction == CSYNC_INSTRUCTION_REMOVE) {
+                        // We need to keep the "delete" items. So we need to un-ignore parent directories
+                        QString parentDir = (*it)->_file;
+                        do {
+                            parentDir = QFileInfo(parentDir).path();
+                            if (parentDir.isEmpty()) {
+                                break;
+                            }
+                            // Find the parent directory in the syncItems vector. Since the vector
+                            // is sorted we can use a lower_bound, but we need a fake
+                            // SyncFileItemPtr needle to compare against
+                            if (!needle)
+                                needle = SyncFileItemPtr::create();
+                            needle->_file = parentDir;
+                            auto parent_it = std::lower_bound(syncItems.begin(), it, needle);
+                            if (parent_it == syncItems.end() || (*parent_it)->destination() != parentDir) {
```

Comment:
Good point.

I'm going to do the check slightly differently.  (because path ends with '/' but we also need to do it for path itself)